feat: add Skills section to AI docs (6 task-based skills) (#2957)
clerkinc/clerk-docs: docs/guides/ai/skills/overview.mdx

---
title: Clerk Skills
description: Install Clerk skills to enhance AI coding agents with authentication expertise.
---

> Skills are installable packages that give AI coding agents specialized knowledge about Clerk. Once installed, your agent can help you add authentication, manage organizations, sync users, and more.

Reviewer: ```suggestion
[Clerk skills](https://github.com/clerk/skills) are installable packages that give AI coding agents specialized knowledge about Clerk. Once installed, your agent can help you add authentication, manage organizations, sync users, and more.
```

## Installation

<Tabs items={["npx (Recommended)", "Manual"]}>
  <Tab>
    Install all Clerk skills:

    ```bash
    npx skills add clerk/skills
    ```

    Or install a specific skill:

    ```bash
    npx skills add clerk/skills --skill adding-auth
    ```

    List available skills before installing:

    ```bash
    npx skills add clerk/skills --list

>     ```

Reviewer: ````suggestion
    List available skills before installing:

    ```bash
    npx skills add clerk/skills --list
    ```

    Install all Clerk skills:

    ```bash
    npx skills add clerk/skills
    ```

    Or install a specific skill:

    ```bash
    npx skills add clerk/skills --skill adding-auth
    ```
````

    Works with 16 agents: Claude Code, Cursor, Windsurf, GitHub Copilot, Codex, Gemini CLI, OpenCode, Amp, Antigravity, Goose, Kilo, Kiro, Roo, Trae, Droid, and Clawdbot.
  </Tab>

  <Tab>
    Clone the skills repository manually:

    ```bash
    git clone https://github.com/clerk/skills ~/.claude/skills/clerk
    ```

    For other agents, check their documentation for the skills directory location.
  </Tab>
</Tabs>

## Available skills

| Skill | Description |
| - | - |
| [adding-auth](/docs/guides/ai/skills/adding-auth) | Add Clerk to Next.js, Express, Remix, or React |
| [customizing-ui](/docs/guides/ai/skills/customizing-ui) | Build custom sign-in and sign-up forms |
| [syncing-users](/docs/guides/ai/skills/syncing-users) | Sync Clerk users to your database via webhooks |
| [testing-auth](/docs/guides/ai/skills/testing-auth) | E2E auth testing with Playwright or Cypress |

> | [managing-orgs](/docs/guides/ai/skills/managing-orgs) | Multi-tenant B2B with organizations and RBAC |

Reviewer: ```suggestion
| [managing-orgs](/docs/guides/ai/skills/managing-orgs) | Multi-tenant B2B with Organizations and Role-Based Access Control |
```
